packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1490,13 +1500,19 @@ class _RenderDecoration extends RenderBox {
         : isOutlineBorder ? -labelHeight * 0.25 : contentPadding.top;
       final double scale = lerpDouble(1.0, _kFinalLabelScale, t)!;
       final double dx;
-      switch (textDirection) {
-        case TextDirection.rtl:
-          dx = labelOffset.dx + label!.size.width * (1.0 - scale); // origin is on the right
-          break;
-        case TextDirection.ltr:
-          dx = labelOffset.dx; // origin on the left
-          break;
+      if(decoration.centerFloatingLabel) {
```

Comment:
Should be `if (` with a space to match the style of the rest of this repository.

Reply:
Resolved and left comments on the design doc.